go/private/extensions.bzl

```diff
@@ -67,6 +67,32 @@ _host_tag = tag_class(
     },
 )
 
+_NOGO_DEFAULT_INCLUDES = ["@@//:__subpackages__"]
+_NOGO_DEFAULT_EXCLUDES = []
+
+_nogo_tag = tag_class(
+    attrs = {
+        "nogo": attr.label(
+            doc = "The nogo target to use when this module is the root module.",
+        ),
+        "includes": attr.label_list(
+            default = _NOGO_DEFAULT_INCLUDES,
+            doc = """
+A Go target is checked with nogo if its package matches at least one of the entries in 'includes'
+and none of the entries in 'excludes'. By default, nogo is applied to all targets in the main
+repository.
+
+Uses the same format as 'visibility', i.e., every entry must be a label that ends with ':__pkg__' or
+':__subpackages__'.
+""",
```

Comment:
Do we want to document the magic value `"all"` here? or it's not intended for usage in bzlmod?

Reply:
I haven't documented it anywhere since I think it's ultimately problematic with Bzlmod: If you use it, the introduction of a new transitive Go dep would be a breaking change. We need to have users explicitly list the deps that they want nogo to be applied to.